File: backend/src/main/java/ebd/etcsLauncher/backend/utils/etcsModuleUtils/ETCSModuleNames.java
package ebd.etcsLauncher.backend.utils.etcsModuleUtils;

import org.jetbrains.annotations.NotNull;

/**
 * Manages the names of supported {@link ebd.etcsLauncher.backend.model.etcsModule.ETCSModule}s.
 *
 * @author Lukas Geyer
 */
public class ETCSModuleNames {

    public static ModuleName stringToModuleName(@NotNull String moduleName) {
        switch(moduleName.toLowerCase()) {
            case "core" -> {return ModuleName.CORE;}
            case "dmi" -> {return ModuleName.DMI;}
            case "graph" -> {return ModuleName.Graph;}
            case "tms", "smarttms" -> {return ModuleName.smartTMS;}
            case "smartlogic" -> {return ModuleName.smartLogic;}
            case "scs", "scenariocontrolserver" -> {return ModuleName.ScenarioControlServer;}
            default -> throw new IllegalArgumentException("No such module exists or string is not mapped: " + moduleName);
        }
    }

    public enum ModuleName {
        CORE,
        DMI,
        Graph,
        smartTMS,
        smartLogic,
        ScenarioControlServer
    }

}

File: frontend/src/constants/ETCSModuleNames.ts
// Module Names
export enum ETCSModuleNames {
    CORE = "CORE",
    DMI = "DMI",
    GRAPH = "Graph",
    SMART_TMS = "smartTMS",
    SMART_LOGIC = "smartLogic",
    SCENARIOCONTROLSERVER = "ScenarioControlServer"
}

export function stringToEnum(moduleNameAsString: string) {
    let moduleName: ETCSModuleNames;

    switch (moduleNameAsString.toLowerCase()) {
        case "core":
            moduleName = ETCSModuleNames.CORE;
            break;
        case "dmi":
            moduleName = ETCSModuleNames.DMI;
            break;
        case "etcs-graph":
        case "graph":
            moduleName = ETCSModuleNames.GRAPH;
            break;
        case "tms":
        case "smarttms":
            moduleName = ETCSModuleNames.SMART_TMS;
            break;
        case "smartlogic":
            moduleName = ETCSModuleNames.SMART_LOGIC;
            break;
        case "scs":
        case "scenariocontrolserver":
            moduleName = ETCSModuleNames.SCENARIOCONTROLSERVER;
            break;
        default:
            throw Error("No such module: " + moduleNameAsString);
    }

    return moduleName;
}
